pkg/controller/registry/resolver/resolver.go

```diff
@@ -20,19 +22,39 @@ import (
 	opregistry "github.com/operator-framework/operator-registry/pkg/registry"
 )
 
+const (
+	runtimeConstraintsFilePath = "runtime_constraints.json"
```

Comment:
I think this could be much nicer. I just don't know what it means for the longer term vision of deppy and olm, etc. If this is still the way we'll want to handle these types of constraints. My thinking right now is to keep the "under-the-covers" way of doing it (baking this file in the downstream operator images) because it's super easy to roll-back and it buys us time to be more deliberate about how we want to handle these constraints in a first-class way. @joelanford @kevinrizza wdyt?